receiver/splunkhecreceiver/receiver.go

```diff
@@ -202,6 +206,23 @@ func (r *splunkReceiver) Start(_ context.Context, host component.Host) error {
 	}
 	mx.NewRoute().HandlerFunc(r.handleReq)
 
+	// set up the ack API handler if the ack extension is present
+	if storageID := r.config.Ack.Extension; storageID != nil {
+		if ext, found := host.GetExtensions()[*storageID]; found {
+			r.ackExt = ext.(ackextension.AckExtension)
+			mx.NewRoute().Path(r.config.Ack.Path).HandlerFunc(r.handleAck)
+		} else {
+			return fmt.Errorf("specified ack extension with id '%q' could not be found", *storageID)
```

Comment:
What's the value of using `%q` over `%s`? `%s` is the more obvious choice for me here, but maybe I'm missing something.

Reply:
This was recommended by @atoulme https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31767/#discussion_r1526499248

Google says use %s on basic strings and double-quote strings as in Go source, use %q.
https://gobyexample.com/string-formatting#:~:text=For%20basic%20string%20printing%20use%20%25s%20.&text=To%20double%2Dquote%20strings%20as,Go%20source%2C%20use%20%25q%20.&text=As%20with%20integers%20seen%20earlier,characters%20per%20byte%20of%20input.

Reply:
Correct me if I'm wrong. but it looks like using `%q` will add double quotes to whatever the given string is. So we should probably remove the single quotes since `%q` will handle it.

Reply:
removed